src/backend/rdtFiber.js

```diff
@@ -1,17 +1,55 @@
-// connect to React DevTools global hook
+// import treeNode, snapshot
 
+let rdtFiberRootNode = null;
 // to hold the array of render events
 const renderEvents = new Map();
 
+// update tree snapshot
+const updateTreeSnapshot = (/* ss */) => {
+  // if rdtFiberRootNode is available, extract the current property
+  if (rdtFiberRootNode) {
+    const { current } = rdtFiberRootNode;
+    // create a Fiber tree with the tree function with current (parameter: current)
+    // in types.js: create Fiber type with all property types
+    // current properties become Tree's fiber data properties on tree
+    // do we also need a ss
+    // TO BE FILLED
+
+  }
+
+}
+
+// connect to React DevTools global hook
 export default () => {
   return () => {
+    // __REACT_DEVTOOLS_GLOBAL_HOOK__ is a global object installed by React Devtools extension's content script that gives access to React fiber nodes
     const rdt = window.__REACT_DEVTOOLS_GLOBAL_HOOK__;
-  };
 
-  // preserve origial onCommitFiberRoot
-  const rdtOnCommitFiberRoot = rdt.onCommitFiberRoot;
-  // intercept the original onCommitFiberRoot with middleware
-  const intercept = function (rdtOnCommit) {
-    
-  }
+    // check if RDT's global object is installed
+    if (!rdt) return;
+
+    // check if application is a React application by checking for a React instance
+    const isReact = rdt.renderers.get(1);
+    if (!isReact) return;
+
+    // get fiberNode information and intercept
+    if (isReact) {
+      rdtFiberRootNode = rdt.getFiberRoots(1).values().next().value;
+
+      // preserve origial onCommitFiberRoot
+      const rdtOnCommitFiberRoot = rdt.onCommitFiberRoot;
```

Comment:
I think this needs to be saved outside of the anonymous function... currently rdtOnCommitFiberRoot is scoped to the anonymous function's execution context, which means it will disappear after the exec context is closed - and we will need it later to undo the monkey patching